File: src/main/scala/com/github/mauricio/postgresql/column/DateDecoder.scala
package com.github.mauricio.postgresql.column

import org.joda.time.LocalDate
import org.joda.time.format.DateTimeFormat

/**
 * User: Maurício Linhares
 * Date: 3/5/12
 * Time: 6:12 PM
 */

object DateDecoder extends ColumnDecoder {

  private val parser = DateTimeFormat.forPattern("yyyy-MM-dd")

  def decode(value: String): LocalDate = {
    this.parser.parseLocalDate(value)
  }
}

File: mysql-async/src/main/java/com/github/mauricio/async/db/mysql/binary/decoder/DateDecoder.kt

package com.github.mauricio.async.db.mysql.binary.decoder

import io.netty.buffer.ByteBuf
import org.joda.time.LocalDate

object DateDecoder : BinaryDecoder {
  override fun decode(buffer: ByteBuf): LocalDate? {
    val result = TimestampDecoder.decode(buffer)

    return result?.toLocalDate()
  }
}
